icu4c/source/test/intltest/caltest.cpp

```diff
@@ -3711,4 +3720,717 @@ void CalendarTest::TestChineseCalendarMapping() {
 
 #endif /* #if !UCONFIG_NO_FORMATTING */
 
+void CalendarTest::TestLocaleTZExtension() {
+    UDate now = Calendar::getNow();
+    static const char* const tz[] = {
```

Comment:
It's ok to have so many test cases, but wouldn't 5 or 10 give you the same coverage?